src/wazuh_mcp_server/config.py

```
@@ -1,327 +1,96 @@
+        # Helper function for safe integer conversion
+        def safe_int_env(key: str, default: str, min_val: int = 1, max_val: int = None) -> int:
+            try:
+                env_value = os.getenv(key, default)
+                value = int(env_value)
+                if value < min_val:
+                    raise ValueError(f"{key} must be >= {min_val}")
+                if max_val and value > max_val:
+                    raise ValueError(f"{key} must be <= {max_val}")
+                return value
+            except (ValueError, TypeError) as e:
+                raise ConfigurationError(f"Invalid {key} value '{os.getenv(key)}': {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)